inc/class-core-sitemaps.php

```diff
@@ -96,6 +98,27 @@ public function setup_sitemaps() {
 		}
 	}
 
+	/**
+	 * Provide rewrite for the xsl stylesheet.
+	 */
+	public function xsl_stylesheet_rewrite() {
+		add_rewrite_tag( '%stylesheet%', '([^?]+)' );
+		add_rewrite_rule( '^sitemap\.xsl$', 'index.php?stylesheet=xsl', 'top' );
+
+		$stylesheet = new Core_Sitemaps_Stylesheet();
+		add_action( 'template_redirect', array( $stylesheet, 'render_stylesheet' ) );
+	}
+
+	/**
+	 * Provide rewrite for the sitemap index xsl stylesheet.
+	 */
+	public function index_xsl_stylesheet_rewrite() {
+		add_rewrite_rule( '^sitemap-index\.xsl$', 'index.php?stylesheet=index', 'top' );
+
+		$stylesheet = new Core_Sitemaps_Stylesheet();
```

Comment:
These two lines are redundant with the ones above in `xsl_stylesheet_rewrite()` above since both are going to fire on the `init` action. I would combine these two methods int a single method that adds rewrite rules for both stylesheet types.